crates/matrix-sdk-ui/src/timeline/inner/state.rs

```diff
@@ -827,7 +827,7 @@ impl TimelineInnerMetadata {
             reactions: Default::default(),
             poll_pending_events: Default::default(),
             fully_read_event: Default::default(),
-            event_should_update_fully_read_marker: Default::default(),
+            has_up_to_date_read_marker_item: Default::default(),
```

Comment:
Again we swapped the meaning here - was it deliberate?

Reply:
Good question, thanks for having me double-check :-)

So this is called transitively at L133 in this file:
https://github.com/matrix-org/matrix-rust-sdk/blob/c207c3e2a815405ed44510a778172d437342d9fa/crates/matrix-sdk-ui/src/timeline/builder.rs#L133-L145

and at L144, we'll call `load_fully_read_event` which will reset it to the correct value, if it loaded a read marker and the event it relates to. Otherwise, we will retry every time we add a new event, until a new fully read marker is received from sync *or* the event referred to by the read marker is added to the timeline.

Reply:
And I've updated it too here so that it's `true` at the beginning as well, because until we see a fully read event, it doesn't make sense to try to update the read marker timeline item, and `load_fully_read_event` will update it accordingly anyways.